### ISSUE TITLE
motion_correction() iterates through file list one frame at a time but requires data with time dimension. 

### ISSUE THREAD

**Issue by gt8mar** [flatironinstitute/CaImAn#1017]
- **Setup info: **: 
 1. Operating system (Linux/macOS/Windows): **Windows**
 
 2. Python version (3.x): **3.10.6**
 
 3. Working environment (Python IDE/Jupyter Notebook/other): **VSCode**
 
 4. Which of the demo scripts you're using for your analysis (if applicable): **demo_motion_correction.ipynb**
 
 5. CaImAn version*:  **I couldn't figure out how to make a params object.**
 
 6. CaImAn installation process (`pip install .`/`pip install -e .`/conda): **the mamba install**

*You can get the CaImAn version by creating a `params` object and then typing `params.data['caiman_version']`. If the field doesn't exist, type N/A and consider upgrading)

- **Describe the issue that you are experiencing**

I am trying to load my own files into the motion_correction function. I have a list of paths to RBG Tiff images (they are actually grayscale so all the color channels are the same) that opens a video with load_movie_chain(). 

When I put my same list of paths into MotionCorrect, I get a series of errors. 
First, I get that:

 m = m[:, indices[0], indices[1]]    #         m = m[indices[0], indices[1]]
IndexError: too many indices for array: array is 2-dimensional, but 3 were indexed

When I dug into this code, I found that in the tiff portion of the function get_file_size(), utilities.py, (line 994) the T and dim variables were actually the size of the rows and columns of one image array. (So T=1280, dim = 1024). The shape of the file was 2 dimensional. This is what is causing the above error, there is no third dimension of m. 

This is happening because motion_correct_rigid(), motion_correction.py (line 278ish), iterates through each frame, taking the time dimension away. It then proceeds to call a bunch of functions that are expecting a time dimension. 

**Is this the result of the cluster (am I doing something wrong by using vscode instead of jupyter) not updating T? Should I load in a giant numpy array of my images? Is there a simple motion_correction script I can use that doesn't thread (or use jupyter)? Is there a folder of images that works in this pipeline?**

**Copy error log below**
\envs\caiman\lib\site-packages\caiman\motion_correction.py", line 2857, in motion_correct_batch_rigid
m = m[:, indices[0], indices[1]]    
IndexError: too many indices for array: array is 2-dimensional, but 3 were indexed

**Comment by kushalkolar** [comment]
I would try just using tiff files with one channel

**Comment by gt8mar** [comment]
I ran with one channel images and got the same exact result. 

The error does not have to do with the number of channels but rather the iteration through frames in the list of "fname".   

When a list of filenames are put into motion_correction.py, it calls motion_correct_rigid(list_of_files), which loads one file and then calls motion_correct_batch_rigid(single_file). motion_correct_batch_rigid() calls get_file_size(), which finds a 2D shape (rows, cols) and returns it as two 1D pieces: T = row, dims = col.   it is expecting a 3D shape (T, rows, cols) which would make sense for the input of a full video, not one frame.

**Comment by j-friedrich** [comment]
Apparently the individual tiff files need to have at least 28 frames (for the default MotionCorrect parameters). I would try concatenating your tiff frames into tiff movies. I attach an example where I first try unsuccessfully to run motion correction on 28 tiff images, and then successfully save them as 1 tiff movie before running motion correction
```
import caiman as cm
from caiman.motion_correction import MotionCorrect
from caiman.utils.utils import download_demo

m=cm.load(download_demo('demoMovie.tif'))
for i in range(28):  # save the first 28 frames as individual tif files
    m[i].save(f'foo{i}.tif')
fnames = list([f'foo{i}.tif' for i in range(28)])

# this doesn't work
mc = MotionCorrect(fnames)
mc.motion_correct(save_movie=True)

# but this works
cm.load(fnames).save('bar.tif')
mc = MotionCorrect('bar.tif')
mc.motion_correct(save_movie=True)
```

**Comment by gt8mar** [comment]
Thank you, that worked! 

TLDR: the load functions expect movie files, not a list of paths to frames. 

I used one-channel images in the end.
